### PR TITLE
lxd-agent: Only unblock systemd once /dev/lxd/sock is started

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

File: lxd-agent/main_agent.go [hunk 1]
```diff
@@ -176,6 +166,16 @@ func (c *cmdAgent) Run(cmd *cobra.Command, args []string) error {
 		}
 	}()
 
+	// Done with early setup, tell systemd to continue boot.
+	// Allows a service that needs a file that's generated by the agent to be able to declare After=lxd-agent
+	// and know the file will have been created by the time the service is started.
+	if os.Getenv("NOTIFY_SOCKET") != "" {
```
Comment (blackboxsw) [inline lxd-agent/main_agent.go:172]: +1 on cloud-init side which experiences a race on Focal VMs when systemd-notify allows cloud-init-local.service to start but sometimes /dev/lxc/sock wasn't present yet.

**Review by tomponline — APPROVED** [review]
Lgtm, but rerunning tests as lots of sftp failures.

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
Hrm tests still quite unhappy, but seemingly unrelated sftp issue.

**Comment by tomponline** [comment]
@stgraber did you see a clear test run after this commit? https://github.com/lxc/lxd/commit/652c745624972b93c2833f450784344ff9aca503

**Comment by stgraber** [comment]
https://jenkins.linuxcontainers.org/job/lxd-github-commit/ is all green

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
Confirmed this issue exists already in main branch, so going to merge this as unrelated to the failures.

**Comment by tomponline** [comment]
Confirmed issue exists in LXD 5.6 tag too.

**Pull request merged**